admin_guide/scheduling/descheduler.adoc

```diff
@@ -0,0 +1,305 @@
+* Pods (link:https://kubernetes.io/docs/tasks/administer-cluster/static-pod/[static and mirror pods] or pods in standalone mode) not associated with a Replica Set (RS), Replication Controller (RC), Deployment, or Job (because these pods are not recreated).
+* Pods associated with xref:../../dev_guide/daemonsets.adoc#dev-guide-daemonsets[DaemonSets].
+* Pods with local storage.
+* Pods subject to xref:../../admin_guide/managing_pods.html#managing-pods-poddisruptionbudget[Pod Disruption Budget (PDB)] 
+are not evicted if descheduling violates the PDB. The pods can be evicted using 
+an [xref:../../admin_guide/out_of_resource_handling.html#out-of-resource-eviction-policyeviction policy].
+
+[NOTE]
+====
+Best efforts pods are evicted before Burstable and Guaranteed pods.
+====
+
+
+[[admin-guide-descheduler-role]]
+== Creating a Cluster Role
+
+To configure the necessary permissions for the descheduler to work in a pod: 
+
+. Create a ../../architecture/additional_concepts/authorization.adoc#roles[cluster role] with the following rules:
++
+[source,yaml]
+----
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: descheduler-cluster-role
+rules:
+- apiGroups: [""]
+  resources: ["nodes"]
+  verbs: ["get", "watch", "list"] <1>
+- apiGroups: [""]
+  resources: ["pods"]
+  verbs: ["get", "watch", "list", "delete"] <2>
+- apiGroups: [""]
+  resources: ["pods/eviction"] <3>
+  verbs: ["create"]
+----
+
+<1> Configures the role to allow viewing nodes.
+<2> Configures the role to allow viewing and deleting pods.
+<3> Allows a node to evict pods bound to itself
+
+. Create the xref:../../admin_guide/service_accounts.adoc#admin-guide-service-accounts[service account] which will be used to run the job:
++
+----
+# oc create sa descheduler-sa -n kube-system
+----
+
+. Bind the cluster role to the service account:
++
+----
+# oc create clusterrolebinding descheduler-cluster-role-binding \
+    --clusterrole=descheduler-cluster-role \
+    --serviceaccount=kube-system:descheduler-sa
+----
+
+[[admin-guide-descheduler-policies]]
+== Creating Descheduler Policies
+
+The descheduler can be configured to remove pods from nodes that violate rules defined by _strategies_ in a YAML policy file. Include a path to the 
+policy file in the xref:admin-guide-descheduler-job[job specification] to apply the specific descheduling strategy.
+
+.Sample descheduler policy file
+[source,yaml]
+----
+[source,yaml]
+----
+apiVersion: "descheduler/v1alpha1"
+kind: "DeschedulerPolicy"
+strategies:
+  "RemoveDuplicates":
+     enabled: false
+  "LowNodeUtilization":
+     enabled: true
+     params:
+       nodeResourceUtilizationThresholds:
+         thresholds: <1>
+           "cpu" : 20
+           "memory": 20
+           "pods": 20
+         targetThresholds: <2>
+           "cpu" : 50
+           "memory": 50
+           "pods": 50
+         numberOfNodes: 3 <3>
+  "RemovePodsViolatingInterPodAntiAffinity": <1>
+     enabled: true
+----
+
```

Comment:
Are the strategies intended to be in one file?

Reply:
Yes, I mean descheduler policy has to be in one file.